### PR TITLE
Fix a wrong error handling code logic in StartupXLOG().

### PR DESCRIPTION
**Pull request by paul-guo-**
Now the code follows PG upstream.

## Here are some reminders before you submit the pull request
- [] Add tests for the change
- [] Document changes
- [] Communicate in the mailing list if needed
- [] Pass `make installcheck`

### CODE REVIEW

File: src/backend/access/transam/xlog.c [hunk 1]
```diff
@@ -7259,7 +7259,7 @@ StartupXLOG(void)
 		 * that as an error, or the database will refuse to start up.
 		 */
 		// WALREP_FIXME: But we should probably do this check in standby mode, too
-		if (StandbyModeRequested || ControlFile->backupEndRequired)
+		if (ArchiveRecoveryRequested || ControlFile->backupEndRequired)
```
Comment (ashwinstar) [inline src/backend/access/transam/xlog.c]: The commit message is very thin and not providing context and description what problem is seen with the existing code, how it lands in that situation, how does making this change solve the same. It makes it hard to review even if we may be making it align to upstream having context around the commit will help in future.

**Comment by paul-guo-** [comment]
@hlinnaka @ashwinstar Sorry, this was found when debugging an issue when doing pg94 stable merge. While looking more at related code and tools I found I might misunderstand something and submitted the PR too earlier. Please ignore this patch at this moment. I'll see how to handle this code change later (Close or modify...)

**Comment by ashwinstar** [comment]
@paul-guo- Do we know more on this issue or should we close it?

**Comment by paul-guo-** [comment]
I think we can close it.

**Pull request closed without merging**